src/datachain/data_storage/warehouse.py

```diff
@@ -220,7 +219,7 @@ def dataset_select_paginated(
         num_yielded = 0
 
         # Ensure we're using a thread-local connection
-        with self.clone() as wh:
+        with self.clone(use_new_connection=True) as wh:
```

Comment:
Was this added because `clone()` could reuse the closed connection by default? Maybe this could be fixed on clickhouse backend instead of having to add this flag everywhere?

Reply:
this flag was added for CH as far as I remember, good catch @skshetry I'm not sure it is needed in this place

(not sure it worth refactoring this flag atm, I'll just revert this flag in this place)